Strange CI error when running unit tests

## 🐛 Bug

A strange CI error (see below) has been showing up in recent CI builds (see #1351, #1354). The stack trace seems to suggest that the moto service is not being stopped at the correct time and is intercepting even HTTP requests when running tests for `HttpFileSystem`.

The error does not occur when I run the tests locally.

```
...
test_http (pipeline.test_file_system.TestGetLocalPath) ... ok
test_local (pipeline.test_file_system.TestGetLocalPath) ... ok
test_s3 (pipeline.test_file_system.TestGetLocalPath) ... ok
test_copy_from_http (pipeline.test_file_system.TestHttpMisc) ... 2022-03-07 09:58:03:rastervision.pipeline.file_system.utils: INFO - Downloading https://raw.githubusercontent.com/tensorflow/models/17fa52864bfc7a7444a8b921d8a8eb1669e14ebd/README.md to /opt/data/tmp/tmpya8fjk2v/http/raw.githubusercontent.com/tensorflow/models/17fa52864bfc7a7444a8b921d8a8eb1669e14ebd/README.md
ERROR

======================================================================
ERROR: test_copy_from_http (pipeline.test_file_system.TestHttpMisc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/raster-vision/raster-vision/tests/pipeline/test_file_system.py", line 450, in test_copy_from_http
    download_if_needed(http_path, self.tmp_dir.name)
  File "/opt/src/rastervision_pipeline/rastervision/pipeline/file_system/utils.py", line 160, in download_if_needed
    fs.copy_from(uri, path)
  File "/opt/src/rastervision_pipeline/rastervision/pipeline/file_system/http_file_system.py", line 106, in copy_from
    with get_file_obj(src_uri) as in_file, open(dst_path,
  File "/opt/src/rastervision_pipeline/rastervision/pipeline/file_system/http_file_system.py", line 31, in get_file_obj
    r = requests.get(uri, stream=True, allow_redirects=True)
  File "/opt/conda/lib/python3.7/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/conda/lib/python3.7/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/responses/__init__.py", line 840, in unbound_on_send
    return self._on_request(adapter, request, *a, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/responses/__init__.py", line 780, in _on_request
    match, match_failed_reasons = self._find_match(request)
  File "/opt/conda/lib/python3.7/site-packages/moto/core/models.py", line 294, in _find_first_match
    for i, match in enumerate(self._matches):
AttributeError: 'RequestsMock' object has no attribute '_matches'

----------------------------------------------------------------------
Ran 162 tests in 14.798s

FAILED (errors=1)
Error: Process completed with exit code 1.
```

## To Reproduce

Steps to reproduce the behavior:

Re-run any CI build.

## Expected behavior

This should not happen.

## Environment

Whatever environment GitHub Actions uses.

## Additional context

N/A

--- a/rastervision_aws_batch/requirements.txt
+++ b/rastervision_aws_batch/requirements.txt
@@ -1,3 +1,3 @@
-boto3==1.17.*
+boto3>=1.9.201
 rastervision_pipeline==0.13
 awscli==1.19.*

--- a/rastervision_aws_s3/requirements.txt
+++ b/rastervision_aws_s3/requirements.txt
@@ -1,4 +1,4 @@
-boto3==1.17.*
+boto3>=1.9.201
 rastervision_pipeline==0.13
 awscli==1.19.*
 tqdm==4.62

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -1,5 +1,5 @@
 flake8==3.5.*
-moto==2.0.10
+moto==3.0.6
 coverage==5.5
 codecov==2.1.11
 yapf==0.22.*

--- a/tests/pipeline/test_file_system.py
+++ b/tests/pipeline/test_file_system.py
@@ -23,13 +23,12 @@ LOREM = """ Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
         mollit anim id est laborum.  """
 
 
+@mock_s3
 class TestMakeDir(unittest.TestCase):
     def setUp(self):
         self.lorem = LOREM
 
         # Mock S3 bucket
-        self.mock_s3 = mock_s3()
-        self.mock_s3.start()
         self.s3 = boto3.client('s3')
         self.bucket_name = 'mock_bucket'
         self.s3.create_bucket(Bucket=self.bucket_name)
@@ -39,7 +38,6 @@ class TestMakeDir(unittest.TestCase):
 
     def tearDown(self):
         self.tmp_dir.cleanup()
-        self.mock_s3.stop()
 
     def test_default_args(self):
         dir = os.path.join(self.tmp_dir.name, 'hello')
@@ -130,13 +128,12 @@ class TestGetLocalPath(unittest.TestCase):
         self.assertEqual(path, '/download_dir/http/bucket/10/25/53')
 
 
+@mock_s3
 class TestFileToStr(unittest.TestCase):
     """Test file_to_str and str_to_file."""
 
     def setUp(self):
         # Setup mock S3 bucket.
-        self.mock_s3 = mock_s3()
-        self.mock_s3.start()
         self.s3 = boto3.client('s3')
         self.bucket_name = 'mock_bucket'
         self.s3.create_bucket(Bucket=self.bucket_name)
@@ -150,7 +147,6 @@ class TestFileToStr(unittest.TestCase):
 
     def tearDown(self):
         self.tmp_dir.cleanup()
-        self.mock_s3.stop()
 
     def test_file_to_str_local(self):
         str_to_file(self.content_str, self.local_path)
@@ -175,13 +171,12 @@ class TestFileToStr(unittest.TestCase):
             file_to_str(wrong_path)
 
 
+@mock_s3
 class TestDownloadIfNeeded(unittest.TestCase):
     """Test download_if_needed and upload_or_copy and str_to_file."""
 
     def setUp(self):
         # Setup mock S3 bucket.
-        self.mock_s3 = mock_s3()
-        self.mock_s3.start()
         self.s3 = boto3.client('s3')
         self.bucket_name = 'mock_bucket'
         self.s3.create_bucket(Bucket=self.bucket_name)
@@ -195,7 +190,6 @@ class TestDownloadIfNeeded(unittest.TestCase):
 
     def tearDown(self):
         self.tmp_dir.cleanup()
-        self.mock_s3.stop()
 
     def test_download_if_needed_local(self):
         with self.assertRaises(NotReadableError):
@@ -221,13 +215,12 @@ class TestDownloadIfNeeded(unittest.TestCase):
             upload_or_copy(local_path, wrong_path)
 
 
+@mock_s3
 class TestS3Misc(unittest.TestCase):
     def setUp(self):
         self.lorem = LOREM
 
         # Mock S3 bucket
-        self.mock_s3 = mock_s3()
-        self.mock_s3.start()
         self.s3 = boto3.client('s3')
         self.bucket_name = 'mock_bucket'
         self.s3.create_bucket(Bucket=self.bucket_name)
@@ -237,7 +230,6 @@ class TestS3Misc(unittest.TestCase):
 
     def tearDown(self):
         self.tmp_dir.cleanup()
-        self.mock_s3.stop()
 
     def test_parse_uri(self):
         s3_path = f's3://{self.bucket_name}/lorem1.txt'
@@ -464,11 +456,10 @@ class TestHttpMisc(unittest.TestCase):
                           lambda: fs.write_bytes(uri, bytes([0x00, 0x01])))
 
 
+@mock_s3
 class TestGetCachedFile(unittest.TestCase):
     def setUp(self):
         # Setup mock S3 bucket.
-        self.mock_s3 = mock_s3()
-        self.mock_s3.start()
         self.s3 = boto3.client('s3')
         self.bucket_name = 'mock_bucket'
         self.s3.create_bucket(Bucket=self.bucket_name)
@@ -480,7 +471,6 @@ class TestGetCachedFile(unittest.TestCase):
 
     def tearDown(self):
         self.tmp_dir.cleanup()
-        self.mock_s3.stop()
 
     def test_local(self):
         local_path = os.path.join(self.tmp_dir.name, self.file_name)

